src/Features/CSharp/Portable/RelatedDocuments/CSharpRelatedDocumentsService.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.RelatedDocuments;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.CSharp.RelatedDocuments;
+
+[ExportLanguageService(typeof(IRelatedDocumentsService), LanguageNames.CSharp), Shared]
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed class CSharpRelatedDocumentsService() : AbstractRelatedDocumentsService<
```

Comment:
language specific logic.  for now, the lang specific logic is just iterating and finding which nodes to bind.  we bind Name nodes (X, X.Y, `X<Y>`, etc.), and member access nodes with a left dotter name (like the `X.Y` in `X.Y.Z()`).